library/manageiq_user.py

```diff
@@ -0,0 +1,269 @@
+
+    def user_update_required(self, user_id, userid, username, group_id, email):
+        """ Returns true if the username, group id or email passed for the user
+            differ from the user's existing ones, False otherwise.
+        """
+        try:
+            url = "{api_url}/users/{user_id}".format(api_url=self.api_url, user_id=user_id)
+            result = self.client.get(url)
+            return True if result['name'] != username or result['current_group_id'] != group_id or result.get('email') != email else False
+        except Exception as e:
+            self.module.fail_json(msg="Failed to get user {userid} details. Error: {error}".format(userid=userid, error=e))
+
+    def update_user(self, user_id, userid, username, group_id, password, email):
+        """Updates the user in manageiq.
+
+        Returns:
+            the created user id, name, created_on timestamp,
+            updated_on timestamp, userid and current_group_id
+        """
+        try:
+            url = '{api_url}/users/{user_id}'.format(api_url=self.api_url, user_id=user_id)
+            resource = {'userid': userid, 'name': username, 'password': password,
+                        'group': {'id': group_id}, 'email': email}
+            result = self.client.post(url, action='edit', resource=resource)
+            self.changed = True
+            return result
+        except Exception as e:
+            self.module.fail_json(msg="Failed to update user {userid}: {error}".format(userid=userid, error=e))
+
+    def create_user(self, userid, username, group_id, password, email):
+        """Creates the user in manageiq.
+
+        Returns:
+            the created user id, name, created_on timestamp,
+            updated_on timestamp, userid and current_group_id
+        """
+        try:
+            url = '{api_url}/users'.format(api_url=self.api_url)
+            resource = {'userid': userid, 'name': username, 'password': password,
+                        'group': {'id': group_id}, 'email': email}
+            result = self.client.post(url, action='create', resource=resource)
+            self.changed = True
+            return result['results']
+        except Exception as e:
+            self.module.fail_json(msg="Failed to create user {userid}: {error}".format(userid=userid, error=e))
+
+    def create_or_update_user(self, userid, username, password, group, email):
+        """ Create or update a user in manageiq.
+
+        Returns:
+            Whether or not a change took place and a message describing the
+            operation executed.
+        """
+        group_id = self.find_group_by_name(group)
+        if not group_id:  # group doesn't exist
+            self.module.fail_json(
+                msg="Failed to create user {userid}: group {group_name} does not exist in manageiq".format(userid=userid, group_name=group))
+
+        user_id = self.find_user_by_userid(userid)
```

Comment:
Heh, this is a confusing line to read.   `userid` is 'fred', `user_id` is 5...

You could start actually using the objects built by `client.collections.foos` instead of passing around ids;
but we're still undecided if we want to depend much on the API client, not for this PR...